src/backend/trpc.ts

```diff
@@ -1,6 +1,34 @@
 import { TRPCError, initTRPC } from "@trpc/server";
-import { createContext } from "./context";
+import { Auth, createContext } from "./context";
 import superjson from "superjson";
+import { UserType } from "@/types/auth";
+
+// Role-based access control type
+type RoleLevel = {
+  user: 0;
+  para: 1;
+  case_manager: 2;
+  admin: 3;
+};
+
+const ROLE_LEVELS: RoleLevel = {
+  user: 0,
+  para: 1,
+  case_manager: 2,
+  admin: 3,
+};
+
+// Function to compare roles
+function hasMinimumRole(auth: Auth, requiredRole: UserType): boolean {
```

Comment:
this function assumes that each role is a superset of the prior one; does this mean a case manager should be able to do what a para does?

Reply:
Yes. Uncommon work flows could be removed from the UI, but I'm thinking in the future there could be an administrator panel that could change database records through trpc calls

Reply:
Yes, @nickvisut for now cascading roles on the backend seems like a good fit for POC/MVP. And as mentioned by Vince, the front-end doesn't do much in the way of auth guards, it just hides ui, but entering a route by hand will render the page, even if the api won't fill the page with data. We can add stories for those use cases as well.